Source/CarthageKit/Project.swift

```diff
@@ -522,7 +522,7 @@ public final class Project {
 			})
 	}
 	
-	public func buildOrderForResolvedCartfile(cartfile: ResolvedCartfile) -> SignalProducer<Dependency<PinnedVersion>, CarthageError> {
+	public func buildOrderForResolvedCartfile(cartfile: ResolvedCartfile, dependenciesToInclude: [String]? = nil) -> SignalProducer<Dependency<PinnedVersion>, CarthageError> {
```

Comment:
>> I think it might make more logical sense to let the caller do the filtering instead.
>
> Actually, my comment doesn't make sense because we need to include the dependencies of any projects we're interested in.

Yes, we need to offer the graph to the caller if the caller should do the filtering.